pkg/util/strategicpatch/patch_test.go

```diff
@@ -55,52 +59,72 @@ type MergeItem struct {
 	SimpleMap         map[string]string
 }
 
-var testCaseData = []byte(`
-strategicMergePatchCases:
-  - description: add new field
-    original:
-      name: 1
-    patch:
-      value: 1
-    result:
-      name: 1
-      value: 1
-  - description: remove field and add new field
+// These are test cases for CreateStrategicMergePatch, used to assert that it
+// generates the correct patch for a given outcome. They are also test cases for
+// StrategicMergePatch, used to assert that applying a patch yields the correct
+// outcome.
+var createStrategicMergePatchTestCaseData = []byte(`
```

Comment:
They were nice, thorough test cases to begin with. It was a treat to modify such a clean implementation.